src/core/app.lua

```diff
@@ -99,7 +99,7 @@ end
 
 -- Run app:methodname() in protected mode (pcall). If it throws an
 -- error app will be marked as dead and restarted eventually.
-local function with_restart (app, method)
+function with_restart (app, method)
```

Comment:
I don't understand why it being local broke this PR? Is it because it was lexically defined after it was used? If so I think the proper solution would be to move the function definition up instead of making it accessible. Nitpick, opinions?